plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -44,190 +38,194 @@ var KustomizePlugin HelmChartInflationGeneratorPlugin
+	cmd.Stderr = stderr
+	env := []string{
+		fmt.Sprintf("HELM_CONFIG_HOME=%s", p.ConfigHome),
+		fmt.Sprintf("HELM_CACHE_HOME=%s/.cache", p.ConfigHome),
+		fmt.Sprintf("HELM_DATA_HOME=%s/.data", p.ConfigHome)}
+	cmd.Env = append(os.Environ(), env...)
+	err := cmd.Run()
 	if err != nil {
-		return err
+		helm := p.h.GeneralConfig().HelmConfig.Command
+		err = errors.Wrap(
+			fmt.Errorf(
+				"unable to run: '%s %s' with env=%s (is '%s' installed?)",
+				helm, strings.Join(args, " "), env, helm),
+			stderr.String(),
+		)
 	}
-	return nil
+	return stdout.Bytes(), err
 }
 
-// useValuesLocal process (merge) inflator config provided values with chart default values.yaml
-func (p *HelmChartInflationGeneratorPlugin) useValuesLocal() error {
-	// not override, merge, none
-	if !(p.ValuesMerge == "none" || p.ValuesMerge == "no" || p.ValuesMerge == "false") {
-		var pValues []byte
-		var err error
-
-		if filepath.IsAbs(p.Values) {
-			pValues, err = ioutil.ReadFile(p.Values)
-		} else {
-			pValues, err = p.h.Loader().Load(p.Values)
-		}
-		if err != nil {
-			return err
-		}
-		chValues := make(map[string]interface{})
-		err = yaml.Unmarshal(pValues, &chValues)
-		if err != nil {
-			return err
+// createNewMergedValuesFile replaces/merges original values file with ValuesInline.
+func (p *HelmChartInflationGeneratorPlugin) createNewMergedValuesFile() (
+	path string, err error) {
+	if p.ValuesMerge == "merge" || p.ValuesMerge == "override" {
+		if err = p.replaceValuesInline(); err != nil {
+			return "", err
 		}
-		if p.ValuesMerge == "override" {
-			err = mergo.Merge(&chValues, p.ValuesLocal, mergo.WithOverride)
-			if err != nil {
-				return err
-			}
-		}
-		if p.ValuesMerge == "merge" {
-			err = mergo.Merge(&chValues, p.ValuesLocal)
-			if err != nil {
-				return err
-			}
-		}
-		p.ValuesLocal = chValues
 	}
-	b, err := yaml.Marshal(p.ValuesLocal)
+	var b []byte
+	b, err = yaml.Marshal(p.ValuesInline)
 	if err != nil {
-		return err
-	}
-	path, err := p.writeValuesBytes(b)
-	if err != nil {
-		return err
+		return "", err
 	}
-	p.Values = path
-	return nil
+	return p.writeValuesBytes(b)
 }
 
-// copyValues will copy the relative values file into the temp directory
-// to avoid messing up with CWD.
-func (p *HelmChartInflationGeneratorPlugin) copyValues() error {
-	// only copy when the values path is not absolute
-	if filepath.IsAbs(p.Values) {
-		return nil
-	}
-	// we must use use loader to read values file
-	b, err := p.h.Loader().Load(p.Values)
+func (p *HelmChartInflationGeneratorPlugin) replaceValuesInline() error {
+	pValues, err := p.h.Loader().Load(p.ValuesFile)
 	if err != nil {
 		return err
 	}
-	path, err := p.writeValuesBytes(b)
-	if err != nil {
+	chValues := make(map[string]interface{})
+	if err = yaml.Unmarshal(pValues, &chValues); err != nil {
 		return err
 	}
-	p.Values = path
-	return nil
+	switch p.ValuesMerge {
+	case "override":
+		err = mergo.Merge(
+			&chValues, p.ValuesInline, mergo.WithOverride)
+	case "merge":
```

Comment:
Should "override" and "merge" be put into constants?

Reply:
willdo

Reply:
done